tests/filecheck/dialects/tensor/invalid_ops.mlir

```diff
@@ -82,3 +82,10 @@ builtin.module {
 
 // CHECK: All inner arrays must be contiguous: [[2 : i64, 3 : i64], [0 : i64, 1 : i64]]
 %res_collapse2 = tensor.collapse_shape %t0 [ [2, 3], [0, 1] ] : tensor<4x1xf32> into tensor<4x1xf32>
+
+// -----
+
+// CHECK: The following constraints were not satisfied:
+// CHECK-NEXT: expected integer >= 0, got -2
+// CHECK-NEXT: All inner arrays must be contiguous: [[-2 : i64, 3 : i64], [0 : i64, 1 : i64]]
```

Comment:
```suggestion
// CHECK-NEXT: All inner arrays must be positive and contiguous: [[-2 : i64, 3 : i64], [0 : i64, 1 : i64]]
```
Could we update this message?

Reply:
or perhaps "non-negative", just worry the message could be confusing if the array was continuous but negative.

Reply:
In retrospect it might be too hard to change this message as it's in a different constraint.